contributors/design-proposals/storage/pv-monitoring-proposal.md

```diff
@@ -0,0 +1,132 @@
+# PV monitoring proposal
+
+Status: Pending
+
+Version: Alpha
+
+Implementation Owner: NickrenREN@ 
+
+## Motivation
+
+For now, kubernetes has no way to monitor the PVs, which may cause serious problems. 
+For example: if volumes are unhealthy, and pods do not know that and still try to get and write data, 
+which will lead to data loss and unavailability of services. 
+So it is necessary to have a mechanism for monitoring PVs and react when PVs have problems.
+
+## Proposal
+
+We can separate the proposal into two parts:
+
+* monitoring PVs and marking them if they have problems
+* reacting to the unhealthy PVs
+
+For monitoring, we may create a controller for it, and each volume plugin should have its own function to check volume health. 
+Controller can call them periodically. The controller also needs to watch node events because local PVs will be unreachable if nodes break down.
+
+For reacting, different kinds of apps may have different methods,we can also create a controller for it. 
+
+At first phase, we may just react to statefulsets with local storage PVs.
+
+## User Experience
+### Use Cases
+
+* If the local PV path is deleted, users should know that and the local PV should be marked and deleted;
```

Comment:
It seems like there's two different reaction controllers being proposed here?

* One for StatefulSets that are using these PVs
* One for local PVs

Could you try to clarify this in the doc and separate out which behaviors will be handled by which controllers?  I think one of the confusing things here is there's multiple levels/layers of reaction and it's not clear how they will all interact with each other.

Reply:
Or maybe this design should just focus on the local PV monitoring part.  And we can leave StatefulSet handling to a different design.

Reply:
yes, reaction needs more discussion, so i will move it to next stage.